Flaky test Test_MultiLevelCache

**Describe the bug**

The following test is failing:

```
--- FAIL: Test_MultiLevelCache (0.08s)
    --- FAIL: Test_MultiLevelCache/[StoreExpandedPostings]_Should_store_on_m2_only (0.00s)
        multilevel_cache_test.go:473: 
            	Error Trace:	/__w/cortex/cortex/pkg/storage/tsdb/multilevel_cache_test.go:473
            	Error:      	Not equal: 
            	            	expected: map[string][][]interface {}{"StoreExpandedPostings":[][]interface {}{[]interface {}{ulid.ULID{0x1, 0x69, 0xd1, 0xdf, 0x90, 0x90, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, []*labels.Matcher{(*labels.Matcher)(0xc000f26ea0)}, []uint8{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}}}}
            	            	actual  : map[string][][]interface {}{}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,26 +1,2 @@
            	            	-(map[string][][]interface {}) (len=1) {
            	            	- (string) (len=21) "StoreExpandedPostings": ([][]interface {}) (len=1) {
            	            	-  ([]interface {}) (len=3) {
            	            	-   (ulid.ULID) (len=16) {
            	            	-    00000000  01 69 d1 df 90 90 00 00  00 00 00 00 00 00 00 00  |.i..............|
            	            	-   },
            	            	-   ([]*labels.Matcher) (len=1) {
            	            	-    (*labels.Matcher)({
            	            	-     Type: (labels.MatchType) 0,
            	            	-     Name: (string) (len=4) "name",
            	            	-     Value: (string) (len=5) "value",
            	            	-     re: (*labels.FastRegexMatcher)(<nil>)
            	            	-    })
            	            	-   },
            	            	-   ([]uint8) (len=100) {
            	            	-    00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
            	            	-    00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
            	            	-    00000020  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
            	            	-    00000030  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
            	            	-    00000040  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
            	            	-    00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
            	            	-    00000060  00 00 00 00                                       |....|
            	            	-   }
            	            	-  }
            	            	- }
            	            	+(map[string][][]interface {}) {
            	            	 }
            	Test:       	Test_MultiLevelCache/[StoreExpandedPostings]_Should_store_on_m2_only
FAIL
FAIL	github.com/cortexproject/cortex/pkg/storage/tsdb	0.896s
```

**To Reproduce**
Steps to reproduce the behavior:
1. `go test -timeout=10s -count=10000 -run=Test_MultiLevelCache ./pkg/storage/tsdb`

--- a/go.mod
+++ b/go.mod
@@ -52,7 +52,7 @@ require (
 	github.com/stretchr/testify v1.10.0
 	github.com/thanos-io/objstore v0.0.0-20241111205755-d1dd89d41f97
 	github.com/thanos-io/promql-engine v0.0.0-20250110162513-14f995518af3
-	github.com/thanos-io/thanos v0.37.3-0.20250107220537-0d426361672a
+	github.com/thanos-io/thanos v0.37.3-0.20250110074750-4ba0ba403896
 	github.com/uber/jaeger-client-go v2.30.0+incompatible
 	github.com/weaveworks/common v0.0.0-20230728070032-dd9e68f319d5
 	go.etcd.io/etcd/api/v3 v3.5.17

